src/components/layout/nav.js

```diff
@@ -558,13 +558,17 @@ const NavDesktop = ({ base, is_ppc, is_ppc_redirect, is_logged_in }) => {
                         has_scrolled={has_scrolled}
                     >
                         <LanguageSwitcherNavDesktop />
-                        <NowrapButton onClick={handleLogin} primary>
-                            <span>{localize('Log in')}</span>
-                        </NowrapButton>
+                        {!hide_sigup_login && (
```

Comment:
You might also want to pass `hide_sigup_login` into `NavRight` to control that the language switcher doesn't slide in to the left upon scrolling.